### PR TITLE
CRM-16355 - reminders in multilingual sites fix

### PR DESCRIPTION
**Pull request by samuelsov**
- [CRM-16355: Reminders in multilingual sites: allow to define the language of the sending](https://issues.civicrm.org/jira/browse/CRM-16355)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@xurizaemon can you look at this!!!

**Comment by JoeMurray** [comment]
Samuel, comme Release Manager ce mois-ci, je suis en train de recruter des gens pour aider à parer l'arriéré de près de 100 PRs, dont certaines remontent à l'été dernier. Je me demande si vous seriez en mesure d'aider AQ autres PRs si je trouve quelqu'un d'AQ ce PR et / ou vos autres ceux # 8538 # 7276?

**Comment by samuelsov** [comment]
@JoeMurray : yes !

**Comment by eileenmcnaughton** [comment]
Oui?

**Comment by JoeMurray** [comment]
Samuel agreed to help QA others' PRs if I get someone to help QA his.

**Comment by JoeMurray** [comment]
@mpnordland willing to try to QA this PR?

File: CRM/Core/BAO/ActionSchedule.php [hunk 1]
```diff
@@ -425,8 +425,12 @@ public static function getRecipientListing($mappingID, $recipientType) {
    * @param $preferred_language
    */
   public static function setCommunicationLanguage($communication_language, $preferred_language) {
-    $config = CRM_Core_Config::singleton();
-    $language = $config->lcMessages;
+    $i18n = CRM_Core_I18n::singleton();
+
+    // FIXME: convert to getLocale when https://issues.civicrm.org/jira/browse/CRM-18362 is fixed
+    global $tsLocale;
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/ActionSchedule.php]: pretty sure I saw a PR merged recently which does this getLocale bit here

**Comment by PalanteJon** [comment]
@samuelsov @JoeMurray I have an affected site in production, I'll QA this!

**Comment by PalanteJon** [comment]
OK, I just ran this through its paces.
First - I confirm that this fixes the issue with sending scheduled reminders on multilingual sites.

Reading through the code, everything makes sense, but two things caught my eye:
* There's a FIXME that says to revisit when CRM-18362 is fixed - which it is.
* I don't understand why there's a [fallback to pre-4.5 mode](https://github.com/civicrm/civicrm-core/pull/8476/files#diff-777f9a255f83c1af5fe6f854699008c0R598).  Is this necessary?  Or is this because the code was backported to a pre-4.5 client and this got left in the PR?

**Comment by samuelsov** [comment]
- about the FIXME, you are right, it's now in Core, i will remove this part
- not sure either, i suppose it's a failsafe attitude... people have the choice of putting their mo files either with or without LC_MESSAGES but the civicrm_l10n files are shipped with the new structure so i guess we don't have to support it for ever... If we remove it there, we should remove it from the constructor to be consistent.

**Comment by samuelsov** [comment]
@PalanteJon : Ok, the patch is ready to be validated again.
I have not fixed the fallback to pre-4.5 mode but i have created two helper function so that at least, there is no duplication of code (setNativeGettextLocale and setPhpGettextLocale) - they are used in the constructor and in the setLocale function. It will be easier to remove the fallback (but it's not quite the goal of this issue).

**Comment by samuelsov** [comment]
The failed tests seems to be unrelated to the patch ("Incomplete Test").

**Comment by colemanw** [comment]
@civicrm-builder retest this please

**Comment by monishdeb** [comment]
Merged on pass :)

**Pull request merged**